src/features/usage-consolidation-accordion/summary-table/summary-table.js

```diff
@@ -5,72 +5,74 @@ import {
   FormattedMessage,
   FormattedNumber,
 } from 'react-intl';
-import getSymbolFromCurrency from 'currency-symbol-map';
 
 import {
   MultiColumnList,
   Button,
   Dropdown,
   DropdownButton,
   DropdownMenu,
-  NoValue,
+  Icon,
   KeyValue,
 } from '@folio/stripes/components';
 
 import { getSummaryTableColumnProperties } from './column-properties';
-import { costPerUse as costPerUseShape } from '../../../constants';
+import {
+  formatCost,
+  formatValue,
+} from '../utilities';
+import {
+  costPerUse as costPerUseShape,
+  entityTypes,
+} from '../../../constants';
 
 const propTypes = {
   contentData: PropTypes.array,
   costPerUseData: costPerUseShape.CostPerUseReduxStateShape.isRequired,
+  costPerUseType: PropTypes.string.isRequired,
   customProperties: PropTypes.object,
+  entityType: PropTypes.string.isRequired,
   id: PropTypes.string.isRequired,
+  onViewTitles: PropTypes.func,
+  year: PropTypes.string.isRequired,
 };
 
 const SummaryTable = ({
   costPerUseData,
   customProperties,
   id,
+  year,
+  costPerUseType,
+  onViewTitles = () => {},
+  entityType,
   ...rest
 }) => {
   const intl = useIntl();
-  const { data } = costPerUseData;
+  const data = costPerUseData.data[costPerUseType];
+  if (!data) {
+    return null;
+  }
 
   const currency = data?.attributes?.parameters?.currency;
-  const currencySymbol = getSymbolFromCurrency(currency) || '';
 
   const {
     cost,
     costPerUse,
     usage,
   } = data?.attributes?.analysis;
 
-  const formatCost = (value) => {
-    return (
-      <FormattedNumber value={value}>
-        {(formattedNumber) => `${currencySymbol}${formattedNumber} (${currency})`}
-      </FormattedNumber>
-    );
-  };
-
-  const formatValue = (value, callback) => {
-    const number = typeof value === 'string' ? Number(value) : value;
-
-    if (!number && number !== 0) {
-      return <NoValue />;
-    }
-
-    const valueToFixed = number.toFixed(2);
-
-    return callback ? callback(valueToFixed) : valueToFixed;
+  const handleViewTitles = (onToggle) => () => {
+    onViewTitles();
+    onToggle();
   };
 
   const formatter = {
-    cost: rowData => formatValue(rowData.cost, formatCost),
-    costPerUse: rowData => formatValue(rowData.costPerUse, formatCost),
+    cost: rowData => formatValue(rowData.cost, (value) => formatCost(currency, value)),
```

Comment:
I will recommend to move this formatting to utilities, a lot of places where it used.